app/components/task-selection/component.js

```diff
@@ -338,7 +366,11 @@ export default class TaskSelectionComponent extends Component {
   onTaskChange(value, options = {}) {
     this._task = value;
 
-    if (!this.project && value?.get("project.id")) {
+    const projectId = value?.get("project.id");
+    if (
+      (!this.project && projectId) ||
+      (projectId && this.project?.id !== projectId)
+    ) {
```

Comment:
@MitanOmar, I covered the reset of the `project` property in this handler, since it will be called anyways. So we can keep our code more *DRY*.